x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/action/StartTrainedModelDeploymentAction.java

```diff
@@ -254,22 +254,28 @@ public static boolean mayAssignToNode(DiscoveryNode node) {
 
         public static final Version VERSION_INTRODUCED = Version.V_8_0_0;
         private static final ParseField MODEL_BYTES = new ParseField("model_bytes");
-        public static final ParseField MODEL_THREADS = new ParseField("model_threads");
-        public static final ParseField INFERENCE_THREADS = new ParseField("inference_threads");
+        public static final ParseField NUMBER_OF_ALLOCATIONS = new ParseField("number_of_allocations");
```

Comment:
Why can't a `ParseField` with a deprecated name be used here as you have done with the request object

Reply:
`TaskParams` is stored in cluster state as part of `TrainedModelAssignment`. If we have a running deployment we want to parse the old names successfully without deprecation warnings. That is why we're trying to read them separately with both the legacy and the new names.

Reply:
I think this is good. The user has no control of "un-deprecating" the previously stored information outside of stopping/starting. I think just allowing it internally like this is fine.